pkg/report/handler_test.go

```diff
@@ -45,3 +57,81 @@ func TestShowHandler(t *testing.T) {
 			status, http.StatusOK)
 	}
 }
+
+type roundTripFunc func(req *http.Request) *http.Response
+
+func (f roundTripFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req), nil
+}
+
+func TestPersistHandler(t *testing.T) {
+	var scenarios = []struct {
+		name                   string
+		fakeResponseStatusCode int
+		getLoadTestsFn         func(action k8stesting.Action) (handled bool, ret runtime.Object, err error)
+		expectedStatusCode     int
+	}{
+		{
+			name:                   "All good",
+			fakeResponseStatusCode: http.StatusOK,
+			getLoadTestsFn: func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
+				return true, &loadtestV1.LoadTest{}, nil
+			},
+			expectedStatusCode: http.StatusOK,
+		},
+		{
+			name:                   "LoadTest not found",
+			fakeResponseStatusCode: http.StatusOK,
+			getLoadTestsFn: func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
+				return true, nil, k8serrors.NewNotFound(loadtestV1.Resource("loadtests"), "loadtest-name")
+			},
+			expectedStatusCode: http.StatusNotFound,
+		},
+		{
+			name:                   "S3 wrong credentials",
```

Comment:
Is it possible to add test case with timeout request from https://github.com/hellofresh/kangal/pull/41/files#diff-68bb18c8eb3eeb398e20ae8dcc42ca2f50628d6d2fe61c83019516b6e846c97dR73 ?